lib/std/fs/File.zig

```diff
@@ -1205,15 +1205,19 @@ pub const Reader = struct {
     pub fn getSize(r: *Reader) SizeError!u64 {
         return r.size orelse {
             if (r.size_err) |err| return err;
-            if (is_windows) {
-                if (windows.GetFileSizeEx(r.file.handle)) |size| {
+            if (is_windows) switch (windows.kernel32.GetFileType(r.file.handle)) {
+                windows.FILE_TYPE_PIPE => {
```

Comment:
```suggestion
                windows.FILE_TYPE_PIPE, window.FILE_TYPE_CHAR => {
```

See https://github.com/ziglang/zig/issues/19994

Reproduction with `File.Reader`:

```
pub fn main() !void {
    var gpa = std.heap.DebugAllocator(.{}){};
    defer std.debug.assert(gpa.deinit() == .ok);
    const allocator = gpa.allocator();

    const stdin = std.fs.File.stdin();
    var reader = stdin.reader(&.{});

    var out: std.Io.Writer.Allocating = .init(allocator);
    defer out.deinit();

    _ = try reader.interface.streamRemaining(&out.writer);
}
```